barretenberg/cpp/src/barretenberg/ultra_honk/ultra_composer.cpp

```diff
@@ -61,23 +58,31 @@ void UltraComposer_<Flavor>::compute_verification_key(const std::shared_ptr<Prov
         verification_key->q_poseidon2_internal = commitment_key->commit(proving_key->q_poseidon2_internal);
     }
 
-    instance->verification_key = std::move(verification_key);
+    return verification_key;
 }
 
 template <IsUltraFlavor Flavor>
-std::shared_ptr<ProverInstance_<Flavor>> UltraComposer_<Flavor>::create_instance(CircuitBuilder& circuit)
+std::shared_ptr<ProverInstance_<Flavor>> UltraComposer_<Flavor>::create_prover_instance(CircuitBuilder& circuit)
 {
     circuit.add_gates_to_ensure_all_polys_are_non_zero();
     circuit.finalize_circuit();
-    auto instance = std::make_shared<Instance>(circuit);
-    commitment_key = compute_commitment_key(instance->proving_key->circuit_size);
+    auto instance = std::make_shared<ProverInstance>(circuit);
+    instance->instance_size = instance->proving_key->circuit_size;
+    commitment_key = compute_commitment_key(instance->instance_size); // hm
```

Comment:
I was thinking that it's not great that the commitment key is computed here but it does depend on the instance_size and this will probably be moved somewhere else in subsequent refactors